### PR TITLE
Handle TVP media in `channel_contributions_forward_pass`

### PR DESCRIPTION
**Pull request by radiokosmos**
## Related Issue

- [ ] Closes #
- [x] Related to #819 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [x] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--998.org.readthedocs.build/en/998/

### CODE REVIEW

**Comment by radiokosmos** [comment]
I am not sure if it's right logic to do time_varying_channel_contributions_forward_pass and need some help with relevant test

File: pymc_marketing/mmm/mmm.py [hunk 1]
```diff
@@ -615,6 +615,68 @@ def channel_contributions_forward_pass(
 
         return idata.posterior_predictive.channel_contributions.to_numpy()
 
+    def time_varying_channel_contributions_forward_pass(
```
Comment (wd60622) [inline pymc_marketing/mmm/mmm.py:618]: Can we just use the channel_contributions_forward_pass method? What is the benefit of having two separate ones?

File: pymc_marketing/mmm/mmm.py [hunk 2]
```diff
@@ -982,8 +1050,13 @@ def get_channel_contributions_forward_pass_grid(
             channel_data = (
                 delta * self.preprocessed_data["X"][self.channel_columns].to_numpy()
             )
+            time_index = (
+                np.arange(0, self.preprocessed_data["X"].shape[0])
+                if self.time_varying_media
+                else None
+            )
```
Comment (wd60622) [inline pymc_marketing/mmm/mmm.py:1057]: Will we know if the dates are correct? Is this only used for the insampel data? What if the `channel_contributions` input are from a subset of the input. We will likely need to use the `infer_time_index` if this is used for new data. This a complaint of mine from the original implementation. If it is always used in the context of the insample, can we not just use the self._time_index variables create in `fit` call.

**Review by wd60622 — COMMENTED** [review]
Just some thoughts to deduplicate logic

File: pymc_marketing/mmm/mmm.py (hunk 1 above)
Comment (radiokosmos) [reply pymc_marketing/mmm/mmm.py:618]: Yeap, makes sense. I'll fix it

File: pymc_marketing/mmm/mmm.py (hunk 2 above)
Comment (radiokosmos) [reply pymc_marketing/mmm/mmm.py:1057]: This is a bit trickier. Should we

**Comment by wd60622** [comment]
Hi @radiokosmos 
Are you still interested in this issue?

**Pull request closed without merging**